test/unit/APIConsumer_unit_test.js

```diff
@@ -1,63 +1,65 @@
-const { networkConfig, autoFundCheck, developmentChains } = require('../../helper-hardhat-config')
-const skipIf = require('mocha-skip-if')
-const chai = require('chai')
-const { expect } = require('chai')
-const BN = require('bn.js')
-const { getChainId } = require('hardhat')
-const {numToBytes32} = require("@chainlink/test-helpers/dist/src/helpers");
-chai.use(require('chai-bn')(BN))
+const { assert, expect } = require("chai")
+const { network, deployments, ethers } = require("hardhat")
+const { developmentChains } = require("../../helper-hardhat-config")
+const { numToBytes32 } = require("@chainlink/test-helpers/dist/src/helpers")
 
-skip.if(!developmentChains.includes(network.name)).
-  describe('APIConsumer Unit Tests', async function () {
+!developmentChains.includes(network.name)
+  ? describe.skip
+  : describe("APIConsumer Unit Tests", async function () {
+      let apiConsumer, linkToken, mockOracle
 
-    let apiConsumer, linkToken, mockOracle
+      beforeEach(async () => {
+        const chainId = network.config.chainId
+        await deployments.fixture(["mocks", "api"])
+        linkToken = await ethers.getContract("LinkToken")
+        linkTokenAddress = linkToken.address
+        additionalMessage = ` --linkaddress  ${linkTokenAddress}`
+        apiConsumer = await ethers.getContract("APIConsumer")
+        mockOracle = await ethers.getContract("MockOracle")
 
-    beforeEach(async () => {
-      const chainId = await getChainId()
-      await deployments.fixture(['mocks', 'api'])
-      const LinkToken = await deployments.get('LinkToken')
-      linkToken = await ethers.getContractAt('LinkToken', LinkToken.address)
-      const networkName = networkConfig[chainId]['name']
-
-      linkTokenAddress = linkToken.address
-      additionalMessage = " --linkaddress " + linkTokenAddress
-
-      const APIConsumer = await deployments.get('APIConsumer')
-      apiConsumer = await ethers.getContractAt('APIConsumer', APIConsumer.address)
-
-      if (await autoFundCheck(apiConsumer.address, networkName, linkTokenAddress, additionalMessage)) {
         await hre.run("fund-link", { contract: apiConsumer.address, linkaddress: linkTokenAddress })
-      }
+      })
 
-      const MockOracle = await deployments.get("MockOracle");
-      mockOracle = await ethers.getContractAt("MockOracle", MockOracle.address);
-    })
+      it("Should successfully make an API request", async () => {
+        const transaction = await apiConsumer.requestVolumeData()
+        const transactionReceipt = await transaction.wait(1)
+        const requestId = transactionReceipt.events[0].topics[1]
```

Comment:
```solidity
const requestId = await apiConsumer.callStatic.requestVolumeData()
```
is cleaner imo, we should add this in the next PR

Reply:
Agreed.